server/middleware/setUpDomainRedirect.test.ts

```diff
@@ -0,0 +1,46 @@
+import express, { Express } from 'express'
+import request from 'supertest'
+import setUpDomainRedirect from './setUpDomainRedirect'
+import config from '../config'
+
+jest.mock('../config', () => ({
+  secondDomain: 'http://example.com',
+  firstDomain: 'http://example.org',
+}))
+
+const setupApp = (): Express => {
+  const app = express()
+  app.use(setUpDomainRedirect())
+
+  app.get('/known', (_req, res, _next) => {
+    res.send('known')
+  })
+
+  return app
+}
+describe('setUpDomainRedirect', () => {
+  it('should redirect to the target domain when in test mode', () => {
+    process.env.NODE_ENV = 'test'
+    const app = setupApp()
+    const targetHost = new URL(config.firstDomain).host
+    return request(app).get('/known').set('host', targetHost).expect(301)
+  })
+
+  it('should not redirect requests already coming from the second domain', () => {
+    process.env.NODE_ENV = 'test'
+    const app = setupApp()
+    const targetHost = new URL(config.secondDomain).host
+    return request(app).get('/known').set('host', targetHost).expect(200)
```

Comment:
Your current implementation is in line with how we do it elsewhere in the codebase, so happy for this to stay as is.

Looking at the examples from [the docs](https://github.com/ladjs/supertest), if we use `done()` `.end()` or `async/await` we can ensure the promise is resolved:

```ts
it('should not redirect requests already coming from the second domain', async () => {
    process.env.NODE_ENV = 'test'
    const app = setupApp()
    const targetHost = new URL(config.secondDomain).host
    const response = await request(app).get('/known').set('host', targetHost)
    expect(response.status).toEqual(200)
})
```